services/director-v2/src/simcore_service_director_v2/modules/comp_scheduler/_scheduler_base.py

```diff
@@ -629,20 +630,28 @@ async def apply(
             msg=f"scheduling pipeline {user_id=}:{project_id=}:{iteration=}",
         ):
             dag: nx.DiGraph = nx.DiGraph()
+
             try:
                 comp_run = await CompRunsRepository.instance(self.db_engine).get(
                     user_id, project_id, iteration
                 )
                 dag = await self._get_pipeline_dag(project_id)
+
                 # 1. Update our list of tasks with data from backend (state, results)
                 await self._update_states_from_comp_backend(
                     user_id, project_id, iteration, dag, comp_run
                 )
-                # 2. Any task following a FAILED task shall be ABORTED
+                # 1.bis get the updated tasks NOTE: we need to get them again as some states might have changed
```

Comment:
bis?

Reply:
<img width="827" height="272" alt="image" src="https://github.com/user-attachments/assets/d5d6bc84-dccc-4aa9-97ec-6f161b7b78be" />
but I will change it